x/staking/types/validator.go

```diff
@@ -501,30 +503,39 @@ func (v Validator) GetOperator() sdk.ValAddress {
 	return addr
 }
 
-// TmConsPubKey casts Validator.ConsensusPubkey to crypto.PubKey
-func (v Validator) TmConsPubKey() (crypto.PubKey, error) {
+// ConsPubKey returns the validator PubKey as a cryptotypes.PubKey.
+func (v Validator) ConsPubKey() (cryptotypes.PubKey, error) {
 	pk, ok := v.ConsensusPubkey.GetCachedValue().(cryptotypes.PubKey)
 	if !ok {
-		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidType, "Expecting crypto.PubKey, got %T", pk)
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidType, "expecting cryptotypes.PubKey, got %T", pk)
 	}
 
-	// The way things are refactored now, v.ConsensusPubkey is sometimes a TM
-	// ed25519 pubkey, sometimes our own ed25519 pubkey. This is very ugly and
-	// inconsistent.
-	// Luckily, here we coerce it into a TM ed25519 pubkey always, as this
-	// pubkey will be passed into TM (eg calling encoding.PubKeyToProto).
-	if intoTmPk, ok := pk.(cryptotypes.IntoTmPubKey); ok {
-		return intoTmPk.AsTmPubKey(), nil
+	return pk, nil
+
+}
+
+// TmConsPubKey casts Validator.ConsensusPubkey to tmcrypto.PubKey.
+func (v Validator) TmConsPubKey() (tmcrypto.PubKey, error) {
```

Comment:
I dont think this is needed. I am not sure if the sdk needs the tmpubkey anywhere but the toTMPubKey functions could just return the proto public key of tendermint, right?

Reply:
see https://github.com/cosmos/cosmos-sdk/pull/7419#discussion_r518688225